pages/book-details/[id].js

```diff
@@ -61,51 +23,33 @@ export default function BookDetailsPage({
     <>
       <StyledDetailedPage>
         <Header />
-        <StyledBookDetail>
-          <StyledImage
-            src={currentBook.cover}
-            height={224}
-            width={150}
-            alt={`Cover Image of ${currentBook.title}`}
-          />
-          <StyledButtonWrapper>
-            <MyLibraryButton
-              onToggle={handleToggleBookmark}
-              id={currentBook.id}
-              isBookmarked={
-                booksInfo.find((bookInfo) => bookInfo.id === currentBook.id)
-                  ?.isBookmarked
-              }
-            />
-            <CurrentlyReadingButton
-              onToggle={handleToggleCurrentlyReading}
-              id={currentBook.id}
-              isCurrentlyReading={
-                booksInfo.find((bookInfo) => bookInfo.id === currentBook.id)
-                  ?.isCurrentlyReading
-              }
-            />
-            <AlreadyReadButton
-              onToggle={handleToggleAlreadyRead}
-              id={currentBook.id}
-              isAlreadyRead={
-                booksInfo.find((bookInfo) => bookInfo.id === currentBook.id)
-                  ?.isAlreadyRead
-              }
-            />
-          </StyledButtonWrapper>
-          <h2>{currentBook.title}</h2>
-          <p>{currentBook.author}</p>
-          <StyledSection>
-            <span>{currentBook.genre}</span>
-            <span>{currentBook.publishYear}</span>
-            <span>{currentBook.pages} Pages</span>
-          </StyledSection>
-          <ReadMoreReadLess description={currentBook.description} />
-        </StyledBookDetail>
+        <BookDetailsCard
+          cover={currentBook.cover}
+          title={currentBook.title}
+          booksInfo={booksInfo}
+          id={currentBook.id}
+          author={currentBook.author}
+          genre={currentBook.genre}
+          publishYear={currentBook.publishYear}
+          pages={currentBook.pages}
+          description={currentBook.description}
+          handleToggleBookmark={handleToggleBookmark}
+          handleToggleCurrentlyReading={handleToggleCurrentlyReading}
+          handleToggleAlreadyRead={handleToggleAlreadyRead}
+          pathName={pathName}
```

Comment:
As described in the `BookDetailCard`, you can pass the object on as a property and use the object destructuring.